opencog/query/PatternTerm.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * PatternTerm.h
+ *
+ * Copyright (C) 2015 OpenCog Foundation
+ * All Rights Reserved
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Created by Jacek Świergocki <jswiergo@gmail.com> July 2015
+ */
+
+#ifndef _OPENCOG_PATTERN_TERM_H
+#define _OPENCOG_PATTERN_TERM_H
+
+#include <vector>
+
+#include <opencog/atomspace/Handle.h>
+#include <opencog/atomspace/Link.h>
+
+namespace opencog {
+
+class PatternTerm;
+typedef std::shared_ptr<PatternTerm> PatternTermPtr;
+typedef std::vector<PatternTermPtr> PatternTermSeq;
+
+class PatternTerm
+{
+	protected:
+		Handle _handle;
+		PatternTermPtr _parent;
+		PatternTermSeq _outgoing;
```

Comment:
Can you add a note explaining how these three members are related to one-another?